### PR TITLE
Add nodeSelector and toleration

### PR DESCRIPTION
**Pull request by marrobi**
- nodeSelector support added by @samisms
- I've added default nodeSelector to prevent attempted deployment to Windows nodes in hybrid clusters
- I've added a toleration so that agent deploys to master node. This taint is added when using ACS, this could alternatively be passed as a default value.

### CODE REVIEW

File: stable/msoms/templates/omsagent-daemonset.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
     release: {{ .Release.Name }}
     app: {{ template "msoms.name" .   }}
    annotations:
-    agentVersion: 1.4.0-45
+    agentVersion: {{ .Values.omsagent.image.tag }}
     dockerProviderVersion: 10.0.0-27
```
Comment (samisms) [inline stable/msoms/templates/omsagent-daemonset.yaml]: Can you change this to 1.0.0-28 ?

**Review by samisms — APPROVED** [review]
Please change the docker-provider version according to comments.

**Comment by marrobi** [comment]
Updated to 10.0.0-28

**Pull request merged**